src/main/java/net/oauth/jsontoken/AsyncJsonTokenParser.java

```diff
@@ -117,8 +119,9 @@ public AsyncJsonTokenParser(
       AsyncVerifierProvider provider =
           asyncVerifierProviders.getVerifierProvider(signatureAlgorithm);
       if (provider == null) {
-        throw new IllegalArgumentException("Signature algorithm not supported: "
```

Comment:
Inside the async parser, we can directly throw the exceptions, right?

Also return immediateFailedFuture here rather than throwing directly.

Reply:
The exceptions weren't directly thrown because I was planning to open another PR to update catching and rethrowing exceptions. For consistency in this PR, I kept the async exceptions thrown to be the same as the sync.

Reply:
In that case, I have updated this to immediately throw our custom exception.

Reply:
When re-throw, please also keep the error message in InvalidJsonTokenException